src/gui/properties/peerlistwidget.cpp

```diff
@@ -400,9 +415,15 @@ void PeerListWidget::updatePeer(const QString &ip, const BitTorrent::TorrentHand
 
 void PeerListWidget::handleResolved(const QString &ip, const QString &hostname)
```

Comment:
Now this has become linear search which is slower than it was. How many peers do you usually have in `m_peerItems`? Because I'm not sure it is worth maintaining another QHash container to speed it up, or we could defer it to another PR...

Reply:
>How many peers do you usually have in m_peerItems?

As many as Maximum # of connections per torrent i guess.
>another QHash container to speed it up

I don't know what you mean?

Reply:
>As many as Maximum # of connections per torrent i guess.

Well. let's defer it later..

>>another QHash container to speed it up
>
>I don't know what you mean?

Something like this: `QHash<QHostAddress, QSet<QStandardItem *>> itemMap;` that will map IP address to a set of `QStandardItem *`. Reason of using it is the lookup time via `QHash::value(key) ` will be much faster than iterating over it.

Reply:
Thanks for the explanation but unfortunately i still can't imagine how this would work... I'm all for optimization but at this point i fear you may be wasting your time trying to explain it to me.

Reply:
I addressed it in PR #11950.